### PR TITLE
Clean up the publicly exported API

### PR DESCRIPTION
**Pull request by dakinggg**
Fixes up a bunch of __all__s

### CODE REVIEW

File: llmfoundry/eval/datasets/__init__.py [hunk 1]
```diff
@@ -8,14 +8,13 @@
...
-                                            stop_sequences_criteria, strip_data,
-                                            tokenizer_needs_prefix_space,
-                                            trim_context)
+
+# isort: off
+from llmfoundry.eval.datasets.utils import (
+    MultiTokenEOSCriteria, convert_tokens_to_tensors, get_continuation_span,
+    get_fewshot_sample_idxs, make_padded_input, stop_sequences_criteria,
+    strip_data, tokenizer_needs_prefix_space, trim_context)
+# isort: on
```
Comment (mvpatel2000) [inline llmfoundry/eval/datasets/__init__.py:17]: why?

**Review by mvpatel2000 — APPROVED** [review]
Can we please add the trailng commas thing? @b-chu has instructions (reisort)

File: llmfoundry/eval/datasets/__init__.py (hunk 1 above)
Comment (dakinggg) [reply llmfoundry/eval/datasets/__init__.py:17]: because isort and yapf disagreed

**Comment by dakinggg** [comment]
@mvpatel2000 Brian said it wasn't as simple as just changing the pyproject and would take a few hours, so I am not going to include that in this PR

@b-chu Not sure if there is a way to automate the alls. Some of the choices of what to reexport are more subjective, but if there is a good way to automate, I'm very open to it.

**Comment by mvpatel2000** [comment]
> @mvpatel2000 Brian said it wasn't as simple as just changing the pyproject and would take a few hours, so I am not going to include that in this PR

Can we prioritize doing this soon? it has frequently come up in PR reviews

**Pull request merged**